go/vt/topo/region.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2017 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package topo
+
+import (
+	"golang.org/x/net/context"
+)
+
+type regionMap struct {
+	cellsToRegions map[string]string
+}
+
+var regions = regionMap{
```

Comment:
I don't see higher level protection from race for this. Better to keep the mutex.

Reply:
The "higher level protection" that I had in mind was that InitRegions is only called once at the startup sequence of DiscoveryGateway.

But considering this more, it's an irrelevant over-optimization that's not actually part of the query serving path, so I added the mutex.